src/tests/baseservices/typeequivalence/simple/Simple.cs

```diff
@@ -268,6 +274,17 @@ private static void LoadInvalidType()
         Console.WriteLine($"-- {typeof(ValueTypeWithInstanceMethod).Name}");
     }
 
+    private static void TestCastsOptimizations()
```

Comment:
I thought that this test needs a fix in the JIT and JIT/EE interface to pass.

Reply:
If we do not optimize isinst that classified as ANY, the test should pass. 

The troubles start when array casts are classified as ANY, then we need to optimize them, but then ANY brings other subtle cases (equivalence, nullable) that are not optimizable.

Reply:
Ok, I see.